app/src/components/core/metaschema/metaSchema.ts

```diff
@@ -16,102 +16,146 @@ module app.core.metaschema {
         static fromJSON(json:any):Metaschema {
             var definitions:Definition[] = [];
 
-            // array contains all names of the elements in the root scope of the json, so they can be referenced by '#'.
-            var rootElements:string[] = [];
+            // array contains the names of all definitions
+            var definitionsNames:string[] = [];
 
             if (json.hasOwnProperty('definitions')) {
-                _.forOwn(json['definitions'], (value:{}, name:string) => {
-                    // we are only looking at object definitions (e.g. 'label' gets sorted out)
+                var definitionsRefs = json['anyOf'];
+                for (var i = 0; i < definitionsRefs.length; i++) {
+                    var name = definitionsRefs[i]['$ref'].substring(("#/definitions/").length);
+                    var value = json['definitions'][name];
                     if (value.hasOwnProperty('type')) {
                         if (value['type'] === 'object') {
-                            var definition:Definition = new Definition(name, Metaschema.cleanupDataschema(value), Metaschema.retrieveAcceptedElements(value, rootElements));
+                            var definition:Definition = new Definition(name, Metaschema.cleanupDataschema(value, json), Metaschema.retrieveAcceptedElements(value, definitionsNames));
 
                             // now continue with all child elements that the current element can accept
-                            _.forEach(Metaschema.retrieveChildElements(value, rootElements), (child:Definition) => {
+                            _.forEach(Metaschema.retrieveChildElements(value, definitionsNames, json), (child:Definition) => {
```

Comment:
The modification of the `definitionsNames` variable is hard to follow. If I didn't miss something, this is potentially the single place where the variable might be read, but it is already written in previous iteration of the loop.